client/my-sites/domains/domain-search-v2/DomainsSearchV2.jsx

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable no-undef */
+/* eslint-disable react/jsx-no-undef */
+
+/**
+ * The DomainSearch* components will be exported from the package so different BUs can assemble the UI as needed.
+ */
+
+export const DomainSearchV2 = () => {
+	const { query, setQuery, filters, setFilters } = useDomainSearch();
+
+	const { exactMatch, hasIncludedTld, recommendedResults, results } = useDomainSuggestions( {
+		query,
+		filters,
+	} );
+
+	const { cart, addToCart, removeFromCart } = useCartManager();
+
+	const [ showFullCart, setShowFullCart ] = useState( false );
+
+	const goToTransferStep = () => {};
+	const goToNextStep = () => {};
+
+	const isDomainOnCart = ( domain, tld ) => {
```

Comment:
Since this is a cross-domain responsibility we need to do it at the front-end as I'm afraid of mixing the concerns in the back-end (i.e., the domains suggestions endpoint knowing about the current users' cart doesn't make sense.)